ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/TestKsqlRestApp.java

```diff
@@ -362,5 +388,9 @@ public Builder withProperties(final Map<String, ?> props) {
     public TestKsqlRestApp build() {
       return new TestKsqlRestApp(bootstrapServers, additionalProps);
     }
+
+    public TestKsqlRestApp buildWithServiceContext(final Supplier<ServiceContext> serviceContext) {
+      return new TestKsqlRestApp(bootstrapServers, additionalProps, serviceContext);
```

Comment:
nit: usually builder patterns would have `withServiceContext(Supplier<ServiceContext>)` - my preference would be to remove the old constructor and have just the new one, but (as with most style things) I'm happy to keep it as is if you feel strongly.

Reply:
I considered that but thought it was weird to have a null-check in the `build()` method. Maybe you had something different in mind? I've implemented what I was thinking earlier, along the lines of your suggestion. Let me know what you think.

Reply:
My idea wasn't to have a null-check, but rather to have a non-final field in the builder with a default.
```java
private Supplier<ServiceContext> serviceContext = () -> defaultServiceContext(bootstrapServers, buildBaseConfig(additionalProps)));
```

Reply:
But this default depends on `bootstrapServers` and `additionalProps`, which can change as a result of calls to `withProperty()` or `withProperties()`. In order to update the default correctly in these situations, we'd have to track whether `withServiceContext()` had been called or not, which is effectively the null-check currently implemented.

Reply:
Resolved offline -- turns out setting the default once at the beginning works fine since `bootstrapServers` and `additionalProps` are final and their actual values won't be picked up until `.get()` is called on the supplier, which happens after `.build()`. Updated to use @agavra 's suggestion. Thanks!